vpr/src/util/ndmatrix_serdes.h

```diff
@@ -0,0 +1,61 @@
+#ifndef NDMATRIX_SERDES_H_
+#define NDMATRIX_SERDES_H_
+
+#include <functional>
+#include "vtr_ndmatrix.h"
+#include "vpr_error.h"
+#include "matrix.capnp.h"
+
+template<size_t N, typename CapType, typename CType>
```

Comment:
Both of these functions need some comments describing:

1. What they do and what their purpose is

2. Internally, how the do it.